client/src/components/EventTypeDialog.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import {ColorPicker} from 'material-ui-color';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import FormLabel from '@material-ui/core/FormLabel';
+
+
+const useStyles = makeStyles((theme) => ({
+
+    btn: {
+        textTransform: 'none',
+    },
+
+    dialog: {
+        padding: 35
+    },
+
+    formControl: {
+        marginTop: 20,
+        marginBottom: 20
+    },
+
+    formControlColor: {
+        marginTop: 30,
+        marginBottom: 10
+    },
+
+    sampleColor: {
+        width: '100%',
+        height: 8,
+        background: theme.palette.primary.main
+    }
+
+}));
+
+export default function EventTypeDialog() {
+        
+    const classes = useStyles();
+
+    const [open, setOpen] = React.useState(false);
+
+    const [values, setValues] = React.useState({
+        title: '',
+        duration: '',
+        url: '',
+        color: ''
+    });
+
+    const handleClickOpen = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const handleInputChange = (prop) => (e) => {
+        setValues({ ...values, [prop]: e.target.value });
+    };
+
+    const handleColorChange = (color) => {
+        setValues({ ...values, 'color': '#'+color.hex})
+    }
+
+    return (
+        <>
+            <Button variant="outlined" color="primary" onClick={handleClickOpen}
+                className={classes.btn}>
+                +&nbsp;&nbsp;New event type
+            </Button>
```

Comment:
A more proper way of handling this is with [Material-UI Button with Icon](https://material-ui.com/components/buttons/#buttons-with-icons-and-label). You can select an icon from [Material UI icons](https://material-ui.com/components/material-icons/)